openmc/statepoint.py

```diff
@@ -534,8 +541,16 @@ def get_tally(self, scores=[], filters=[], nuclides=[],
 
                 # Iterate over the Filters requested by the user
                 for filter in filters:
-                    if filter not in test_tally.filters:
-                        contains_filters = False
+                    contains_filters = False
+
+                    # Test if requested filter is a subset of any of the test
+                    # tally's filters and if so continue to next filter
+                    for test_filter in test_tally.filters:
+                        if test_filter.is_subset(filter):
+                            contains_filters = True
+                            break
+
+                    if not contains_filters:
                         break
```

Comment:
I don't know how others feel about it, but I've actually come to like Python's relatively little known `for/else` construct for this type of logic:
```Python
for test_filter in test_tally.filters:
    if test_filter.is_subset(filter):
        break
else:
    break
```

The else clause gets executed only when the for clause is not broken out of, if that makes sense. Thus it removes the need to have the `contains_filters` logical.